src/zerialize.ts

```diff
@@ -153,7 +153,57 @@ const zerializers = {
     );
     return { type: "number", ...checks };
   },
-  ZodString: () => ({ type: "string" }),
+  ZodString: (def) => {
+    const checks = def.checks.reduce(
+      (o, check) => ({
+        ...o,
+        ...(check.kind == "min"
+          ? { min: check.value }
+          : check.kind == "max"
+          ? { max: check.value }
+          : check.kind == "length"
+          ? { length: check.value }
+          : check.kind == "startsWith"
+          ? { startsWith: check.value }
+          : check.kind == "endsWith"
+          ? { endsWith: check.value }
+          : check.kind == "includes"
+          ? { includes: check.value, position: check.position }
+          : check.kind == "regex"
+          ? {
+              regex: check.regex.source,
+              ...(check.regex.flags ? { flags: check.regex.flags } : {}),
+            }
+          : check.kind == "ip"
+          ? { stringType: "ip", version: check.version }
+          : check.kind == "datetime"
+          ? {
+              stringType: "datetime",
+              ...(check.offset ? { offset: check.offset } : {}),
+              ...(typeof check.precision === "number"
+                ? { precision: check.precision }
+                : {}),
+            }
+          : check.kind == "email"
+          ? { stringType: "email" }
+          : check.kind == "url"
+          ? { stringType: "url" }
+          : check.kind == "emoji"
+          ? { stringType: "emoji" }
+          : check.kind == "uuid"
+          ? { stringType: "uuid" }
+          : check.kind == "cuid"
+          ? { stringType: "cuid" }
+          : check.kind == "cuid2"
+          ? { stringType: "cuid2" }
+          : check.kind == "ulid"
+          ? { stringType: "ulid" }
+          : {}),
```

Comment:
I just realised that the naming I gave wasn't great, i.e. I'd prefer sticking close to zod in general, thus can we go to `kind` from `stringType`.

And I'd also think this here could be deduped a bit, e.g.

```ts
const STRING_KINDS = ['email', 'url', ...];

...
  STRING_KINDS.includes(check.kind) ? { kind: check.kind } : {}
...
```

Wdyt?

Reply:
Sure, was thinking you might not want that redundancy. I switched it to a `Set` for somewhat better performance.